equinox/nn/pool.py

```diff
@@ -0,0 +1,238 @@
+from typing import Sequence, Tuple, Union, Callable, Optional
+import einops
+
+import jax.lax as lax
+import jax.numpy as jnp
+import jax.random
+import numpy as np
+
+from ..custom_types import Array
+from ..module import Module, static_field
+
+
+class Pool(Module):
+    """ General N-dimensional downsampling over a sliding window. """
+
+    operation: Callable[[Array], Array] = static_field()
+    num_spatial_dims: int = static_field()
+    kernel_size: Union[int, Sequence[int]] = static_field()
+    stride: Union[int, Sequence[int]] = static_field()
+    padding: Union[int, Sequence[int]] = static_field()
+    init: float = static_field()
+
+    def __init__(
+        self,
+        operation: Callable[[Array], Array],
+        num_spatial_dims: int,
+        kernel_size: Union[int, Sequence[int]],
+        stride: Union[int, Sequence[int]] = 1,
+        padding: Union[int, Sequence[int], Sequence[tuple[int, int]]] = 0,
+        **kwargs,
+    ):
+        """**Arguments:**
+        - `operation`: The operation applied to the inputs of each window.
+        - `num_spatial_dims`: The number of spatial dimensions.
+        - `kernel_size`: The size of the convolutional kernel.
+        - `stride`: The stride of the convolution.
+        - `padding`: The amount of padding to apply before and after each
+            spatial dimension.
+        """
+        super().__init__(**kwargs)
+
+        self.operation = operation
+        self.init = 0.0
+        if operation is lax.max:
+            self.init = -jnp.inf
```

Comment:
Make `init` an input, don't try to guess based on special values. Essentially `Pool` should make sense as a standalone operation.